File: StudentAffairsSystem/StudentAffairsSystem/StudentAffairsSystem/ClientApp/src/app/models/StudentFormViewModel.ts
import { Guid } from 'guid-typescript';

export interface StudentFormViewModel {
    id: string;
    gender: string;
    name: string;
    birthDate: Date;
    adress: string;
    phone: string;
    emailAddress: string;
    photo: string;
    classId: string;
    class: string;
}
export interface ClassViewModel {
    id: string;
    name: string;
}
export class StudentEditor implements StudentEditor
{

    public id: string='';
    public gender: string='';
    public name: string='';
    public birthDate: Date=new Date();
    public adress: string='';
    public phone: string='';
    public emailAddress: string='';
    public photo: string='';
    public classId: string='';
    public class: string;
    public classList:ClassViewModel[]=[{ id:'11', name:"A"},{ id:'22', name:"B"}]

 constructor()
 {
     this.id = Guid.create().toString();
 }
 
}

File: StudentAffairsSystem/StudentAffairsSystem/StudentAffairsSystem/ViewModels/StudentFormViewModel.cs
// =============================
// Email: info@ebenmonney.com
// www.ebenmonney.com/templates
// =============================

using System;
using System.ComponentModel.DataAnnotations;

namespace StudentAffairsSystem.ViewModels
{
    public class StudentFormViewModel {
       
        public Guid? Id { get; set; }
        public string Gender { get; set; }
        [MaxLength(256)]
        [Required]
        public string Name { get; set; }
        [Required]
        public DateTime BirthDate { get; set; }
        [MaxLength(500)]
        public string Adress { get; set; }
        [MaxLength(40)]
        public string Phone { get; set; }
        [MaxLength(256)]
        public string EmailAddress { get; set; }
        [MaxLength(500)]
        public string Photo { get; set; }

       
        public Guid? ClassId { get; set; }

        public string Class { get; set; }
    }
}
